src/tdlib/chat_history.rs

```diff
@@ -114,17 +126,25 @@ impl ChatHistory {
             .map(|m| m.id())
             .unwrap_or_default();
 
-        self.set_loading(true);
+        imp.is_loading.set(true);
 
         let result =
-            functions::get_chat_history(chat_id, oldest_message_id, 0, 20, false, client_id).await;
+            functions::get_chat_history(chat_id, oldest_message_id, 0, limit, false, client_id)
+                .await;
+
+        imp.is_loading.set(false);
 
-        if let Ok(enums::Messages::Messages(result)) = result {
-            let messages = result.messages.into_iter().flatten().collect();
-            self.append(messages);
+        match result {
+            Ok(enums::Messages::Messages(data)) => {
+                let messages = data.messages.into_iter().flatten().collect();
+                self.append(messages);
+            }
+            Err(e) => {
+                return Err(ChatHistoryError::Tdlib(e));
+            }
```

Comment:
We could avoid the nesting and make the code a bit more readable. Still have to remove the closing `}`
```suggestion
        let enums::Messages::Messages(data) = result.map_err(ChatHistoryError::Tdlib)?;

        let messages = data.messages.into_iter().flatten().collect();
        self.append(messages);
```